state/in_memory_store.go

```diff
@@ -20,7 +20,7 @@ type InMemoryStore struct {
 // configStorage is the storage for the configuration with its own mutex lock
 type configStorage struct {
 	mu   sync.RWMutex
-	conf *config.Config
+	conf config.Config
```

Comment:
since this defines a configStorage object, which is effectively a config + mutex, would it make sense to embed the config instead? That way you won't need `conf.conf` calls.

Reply:
I always forget about embedding! Great idea, cleans up the `conf.conf` and makes readability a lot better